src/EditorFeatures/Core/Implementation/IntelliSense/Completion/AsyncCompletionService.cs

```diff
@@ -51,12 +60,21 @@ internal class AsyncCompletionService : ForegroundThreadAffinitizedObject, IAsyn
 
             _autoBraceCompletionChars = autoBraceCompletionChars;
             _autoBraceCompletionCharSet = new Dictionary<IContentType, ImmutableHashSet<char>>();
+
+            _legacyCompletionEnabledCookie = featureServiceFactory.GlobalFeatureService.GetCookie(nameof(LegacyRoslynCompletion));
```

Comment:
I think this should be scoped to the view, as in, you should check for existence of flag for the view only.  If this is a global flag, then it would a riskier change because it affects the entire session of Roslyn (beyond the currently opened workspace/document).  If LSP fails to dispose (exceptions occurring or what not), then any files that he user opens after LiveShare session ends will not get completion from Roslyn at all.  If you scope it to the TextView, then even if I (LSP) fail to dispose, it's only scoped to that text view, which will eventually be closed.  This way, we won't have any potential for lingering side effects.

Reply:
instead of using a global cookie, you'll need to check the feature state for each view. 
If TryGetController may be called multiple times, cache the per-view cookie